specs/src/recovery.md

```diff
@@ -0,0 +1,525 @@
+combatting sybil attacks. 
+
+The more voting power that has signaled that they have a transaction, then the
+more difficult it is to stop the distribution of that transaction.
+
+### Synthesis
+
+Alone, none of the mechanisms listed above are sufficient for preventing sybil
+attacks. However, when we combine them we can meet our original goal:
+
+>Its important that new broadcasting mechanisms do not introduce a lower barrier
+>to halting the network than standard proof of stake already does.
+
+- If parity data is also gossiped, then the acceptable number of failures is
+  raised relative to the erasure encoding factor.
+- If an overlay network is added, then validators are able to avoid sybil
+  attacks from non-validators
+- If some portion of the proposed block is already distributed to a majority of
+  validators, then it has already been sufficiently distributed to bypass sybil.
+
+## Messages
+
+### Commitment
+
+```proto
+// TxMetaData contains all information needed to commit to a transaction. Start
+// and end are used to insert the transaction into the protobuf encoded version
+// of the block.
+message TxMetaData {
+  bytes  hash  = 1;
+  uint32 start = 2;
+  uint32 end   = 3;
+}
+
+// CompactBlock commits to the transaction included in a proposal. It enalbes
+// clients to reuse already downloaded blobs instead of gossiping them all again
+// during the block propagation.
+message CompactBlock {
+  bytes                     bp_hash      = 1;
+  repeated TxMetaData       blobs        = 2;
+  bytes                     signature    = 3;
+  tendermint.types.Proposal proposal     = 4;
+  uint32                    last_length  = 5;
+  repeated bytes            parts_hashes = 6;
+}
+```
+
+The compact block is signed over by the proposer, and verified by converting to
+signbytes, and verifying using the proposer's public key to verify the included
+signature. This siganture is separate from the proposal signature as it is
+purely related to block propagation, and not meant to be part of the proposal.
+This allows for block propagation to be backwards compatible with older
+implementations.
+
+The `BlobMetaData` contains the hash of the PFB for the blob transaction that it
+commits to, alongside the `size`. `size` is the length of the blobTx in bytes.
+
+The `bp_hash` is generated by taking the merkle root over of each of the blob
+transactions in `BlobMetaData` and `Have` messasges.
+
+Verification:
+
+- The signature MUST be valid and from the expected proposer for that height and
+  round.
+
+### Have
+
+```proto
+// PartMetaData proves the inclusion of a part to the block.
+message PartMetaData {
+  uint32 index = 1;
+  bytes  hash  = 2;
+}
+
+// HaveParts is a message sent by a peer to inform that it has or will receive a
+// part of a block.
+message HaveParts {
+  int64                 height = 1;
+  int32                 round  = 2;
+  repeated PartMetaData parts  = 3;
```

Comment:
Is it important to cap this s.t. a HaveParts with too many indices would immediately be rejected? 🤔